server/routerlicious/packages/services/src/mongoExceptionRetryRules/IMongoExceptionRetryRule.ts

```diff
@@ -0,0 +1,27 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export interface IMongoExceptionRetryRule {
+	ruleName: string;
+	match: (error: any) => boolean;
+	shouldRetry(): boolean;
+}
+
+export abstract class BaseMongoExceptionRetryRule implements IMongoExceptionRetryRule {
+	public abstract match(error: any): boolean;
+	protected abstract defaultDecision: boolean;
```

Comment:
nit: personal pref would be `defaultRetryDecision` for clarity, but not a necessary refactor

Reply:
No problem. changed the name in 15d3cb583b16660927fc939564723dbb325cd2b4